centreon/src/Core/HostCategory/Application/UseCase/AddHostCategory/AddHostCategory.php

```diff
@@ -0,0 +1,109 @@
+*
+*/
+
+declare(strict_types=1);
+
+namespace Core\HostCategory\Application\UseCase\AddHostCategory;
+
+use Centreon\Domain\Contact\Contact;
+use Centreon\Domain\Contact\Interfaces\ContactInterface;
+use Centreon\Domain\Log\LoggerTrait;
+use Core\Application\Common\UseCase\CreatedResponse;
+use Core\Application\Common\UseCase\ErrorResponse;
+use Core\Application\Common\UseCase\ForbiddenResponse;
+use Core\Application\Common\UseCase\InvalidArgumentResponse;
+use Core\Application\Common\UseCase\PresenterInterface;
+use Core\HostCategory\Application\Exception\HostCategoryException;
+use Core\HostCategory\Application\Repository\ReadHostCategoryRepositoryInterface;
+use Core\HostCategory\Application\Repository\WriteHostCategoryRepositoryInterface;
+use Core\HostCategory\Domain\Model\HostCategory;
+use Core\HostCategory\Domain\Model\NewHostCategory;
+use Core\HostCategory\Infrastructure\API\AddHostCategory\AddHostCategoryPresenter;
+
+final class AddHostCategory
+{
+    use LoggerTrait;
+
+    public function __construct(
+        private WriteHostCategoryRepositoryInterface $writeHostCategoryRepository,
+        private ReadHostCategoryRepositoryInterface $readHostCategoryRepository,
+        private ContactInterface $user
+    ) {
+    }
+
+    /**
+     * @param AddHostCategoryRequest $request
+     * @param AddHostCategoryPresenter $presenter
+     */
+    public function __invoke(AddHostCategoryRequest $request, PresenterInterface $presenter): void
+    {
+        try {
+            if (! $this->user->hasTopologyRole(Contact::ROLE_CONFIGURATION_HOSTS_CATEGORIES_READ_WRITE)) {
+                $this->error('User doesn\'t have sufficient right to see host categories', [
+                    'user_id' => $this->user->getId(),
+                ]);
+                $presenter->setResponseStatus(
+                    new ForbiddenResponse(HostCategoryException::addNotAllowed()->getMessage())
+                );
+            } elseif ($this->readHostCategoryRepository->existsByName(trim($request->name))) {
+                $this->error('Host category name already exists', [
+                    'hostcategory_name' => trim($request->name),
+                ]);
+                $presenter->setResponseStatus(
+                    new InvalidArgumentResponse(HostCategoryException::hostNameAlreadyExists()->getMessage())
+                );
+            } else {
+                $newHostCategory = new NewHostCategory(trim($request->name), trim($request->alias));
+                $newHostCategory->setComment($request->comment ? trim($request->comment) : null);
+
+                $hostCategoryId = $this->writeHostCategoryRepository->add($newHostCategory);
+                $hostCategory = $this->readHostCategoryRepository->findById($hostCategoryId);
+
+                $presenter->present(
+                    new CreatedResponse($hostCategoryId, $this->createResponse($hostCategory))
+                );
+            }
```

Comment:
Several questions business related : 
- No isAdmin check ?
- No acl_resources added for a contact ?

Reply:
- user->hasTopologyRole return always true when user is admin
- based on UI behaviour, ACLs are not modified at the creation of a new host category